src/main/java/graphql/schema/FieldDataFetcher.java

```diff
@@ -2,52 +2,36 @@
 
 
 import java.lang.reflect.Field;
-import java.util.Map;
+
+import static graphql.Assert.assertNotNull;
 
 /**
  * Fetches data directly from a field.
  */
 public class FieldDataFetcher implements DataFetcher {
 
-    /**
-     * The name of the field.
-     */
     private final String fieldName;
 
-    /**
-     * Ctor.
-     *
-     * @param fieldName The name of the field.
-     */
     public FieldDataFetcher(String fieldName) {
+        assertNotNull(fieldName, "`name` can't be null");
         this.fieldName = fieldName;
     }
 
     @Override
-    public Object get(DataFetchingEnvironment environment) {
-        Object source = environment.getSource();
+    public Object get(DataFetchingEnvironment env) {
+        Object source = env.getSource();
         if (source == null) return null;
-        if (source instanceof Map) {
```

Comment:
This PR includes a [`MapDataFetcher`](https://github.com/dminkovsky/graphql-java/blob/7f483eff91416a67822e615035157453b3b87e24/src/main/java/graphql/schema/MapDataFetcher.java)